src/tools/isabelle_executor.py

```diff
@@ -208,12 +216,16 @@ def run_next(self) -> bool:
         self.current_stmt = stmt
         self.line_num += 1
         try:
-            self._run_stmt_on_isabelle_server(stmt)
+            self._run_stmt_on_isabelle_server(stmt, proof_search_mode)
         except:
-            if not self.suppress_error_log:
-                logger.error(f"Got an exception while running '{stmt}' on isabelle. File name: {self.main_file}")
-                logger.exception(f"Exception Log")
-            raise
+            if proof_search_mode:
+                if not self.suppress_error_log:
+                    logger.error(f"Got an exception while running '{stmt}' on isabelle. File name: {self.main_file}")
+                    logger.exception(f"Exception Log")
+                raise
+            else:
+                # If we're not in proof search mode, we can assume any errors are expected
+                pass
```

Comment:
added a temporary fix to the newline-parsing problem. We do so by differentiating between two different execution modes: proof_search_mode and non-proof_search_mode. When we are executing Isabelle tactics not in proof search mode, this is usually because we are executing an existing compilable file.

In this case, we don’t expect any errors to be thrown. If we get errors, it’s probably just because a tactic was split up over multiple lines in the file. In this case, we’ll just put the failed tactic in a buffer, and continue adding to the buffer until the tactic is complete.

When querying ChatGPT, we should use proof_search_mode. When running functions like run_to_finish(), run_till_line_num(), etc., we won’t use proof_search_mode.